### PR TITLE
update dockerfile - maint/pre-hnn-core branch

### PR DESCRIPTION
**Pull request by jashlu**

### CODE REVIEW

File: installer/docker/Dockerfile [hunk 1]
```diff
@@ -83,17 +90,18 @@ LABEL org.label-schema.build-date=$BUILD_DATE \
       org.label-schema.vcs-ref=$VCS_REF \
       org.label-schema.schema-version=$VCS_TAG
 
-# install NEURON
-RUN sudo pip install NEURON
+# install hnn-core
+RUN pip install hnn-core
+
```
Comment (rythorpe) [inline installer/docker/Dockerfile:95]: We shouldn't need hnn-core on this branch.

File: installer/docker/Dockerfile [hunk 2]
```diff
@@ -83,17 +90,18 @@ LABEL org.label-schema.build-date=$BUILD_DATE \
...
+RUN sudo apt-get update && \
+    sudo apt-get install -y neuron-dev
```
Comment (rythorpe) [inline installer/docker/Dockerfile:97]: Out of curiosity, is this preferred over doing a pip installation of NEURON?

File: installer/docker/Dockerfile (hunk 1 above)
Comment (jashlu) [reply installer/docker/Dockerfile]: Gotcha, thanks!

File: installer/docker/Dockerfile (hunk 2 above)
Comment (jashlu) [reply installer/docker/Dockerfile:94]: Just from experience I was having problems installing neuron with pip3, and so from some online resources (https://command-not-found.com/nrnivmodl) I tried an alternative and it worked

Comment (rythorpe) [reply installer/docker/Dockerfile:94]: Hmm, I've never run into issues using pip except on Windows (without WSL). NEURON doesn't currently distribute a python wheel for windows, but it should work for WSL, native linux, and macOS.

Comment (jashlu) [reply installer/docker/Dockerfile:94]: This is the error I run into 

```
9.505 Processing triggers for libc-bin (2.31-0ubuntu9.12) ...
9.545 Cloning into '/home/hnn_user/hnn_source_code'...
10.82 nrnivmodl mod
10.82 make: nrnivmodl: Command not found
10.82 make: *** [Makefile:11: x86_64/special] Error 127

```

Comment (jashlu) [reply installer/docker/Dockerfile:94]: To further clarify, I run into this error attempting to build the image on my mac

Comment (jasmainak) [reply installer/docker/Dockerfile:94]: I guess the "barebone" mac install in a docker is not the same as an actual mac computer ... there may be some compilers etc missing

**Comment by rythorpe** [comment]
@jashlu thanks for your work on this. Is it ready for a final review, or is there still more to do?

**Comment by ntolley** [comment]
I think the final decision was to just stick with the docker build for `maint/pre-hnn-core` as the official version we recommend.

@dylansdaniels perhaps let's discuss in our next meeting updating the website documentation to make this clearer? right now people will be attempting to install a version of HNN-GUI that is fairly unstable as `hnn-core` continues to change

**Comment by rythorpe** [comment]
Just to clarify @jashlu, we went ahead and merged your recent docker build changes for `master` since you already had it working, but still want to encourage users to use the docker build on `maint/pre-hnn-core`.

**Comment by jashlu** [comment]
> Just to clarify @jashlu, we went ahead and merged your recent docker build changes for `master` since you already had it working, but still want to encourage users to use the docker build on `maint/pre-hnn-core`.

Understood, thanks for the explanation. I will hold off on merging just for a bit while I work through adding the script to simplify one step in the xserver installation, I will tag you guys for re-review when it is done, should be done soon, thanks!

**Comment by rythorpe** [comment]
I'll try to test this out on my windows machine this weekend. Feel free to bug me early next week if I don't get back to you soon @jashlu!

**Comment by jashlu** [comment]
> I'll try to test this out on my windows machine this weekend. Feel free to bug me early next week if I don't get back to you soon @jashlu!

That would be greatly appreciated, have a great weekend!

**Comment by rythorpe** [comment]
@dylansdaniels I just noticed your [comment](https://github.com/jonescompneurolab/hnn/pull/337#issuecomment-1921901401) in #337. Are you fetching from this branch? If not, I believe this the branch we (`maint/pre-hnn-core`) that we want to be testing and using in the future.

**Comment by dylansdaniels-berkeley** [comment]
So I was moreso referring to how if you follow the docker instructions in the pre-hnn-core branch, it will have you clone from hnn.git, which will put you on master branch. And then if you proceed to the next step in the instructions, it'll direct you to ./scripts/configure_vcxsrv.sh, but you won't be on the branch that has that file.

File: installer/docker/README.md [hunk 3]
```diff
@@ -1,81 +1,109 @@
...
-### Without SSH (only recommended for Linux)
 
-```bash
-export USE_SSH=0
-docker exec --env SYSTEM_USER_DIR="$HOME" --env DISPLAY=":0" -u "$(id -u)" hnn_container /home/hnn_user/start_hnn.sh
+### Part 4 Configure and Run HNN Application
+1. Open up a new Ubuntu terminal application, referenced in Step 3 of Part 1. (Note: ensure that the Ubuntu terminal is being used and not the Windows Powershell)
+2. We will now download the source code that starts up the GUI. Paste this command to create a copy of the source code on your computer.
+```
+git clone https://github.com/jonescompneurolab/hnn.git
```
Comment (rythorpe) [inline installer/docker/README.md:68]: @dylansdaniels I see, so this line here needs to be changed to something like:
```suggestion
git clone --single-branch --branch maint/pre-hnn-core https://github.com/jonescompneurolab/hnn.git
```

Comment (dylansdaniels) [reply installer/docker/README.md:68]: Yes exactly!

Comment (jashlu) [reply installer/docker/README.md]: Thank you for the review, change commited

**Comment by rythorpe** [comment]
Any other recommendations and/or concerns here @dylansdaniels?

**Comment by dylansdaniels** [comment]
Sorry for the delay @rythorpe. That covers it in terms things I bumped into doing a fresh install on Windows 11

**Review by rythorpe — COMMENTED** [review]
Thanks @dylansdaniels for testing this out! @jashlu, as soon as you feel this is ready on your end, please update the title of this PR to "[MRG]..." to signal that it's on the final stage of review.

**Comment by rythorpe** [comment]
I just realized we never completed this. Since I'm pretty sure it's ready to go, let's go ahead and merge. Thanks @jashlu!

**Pull request merged**